metrics/exporter.go

```diff
@@ -82,17 +105,48 @@ func UpdateExporter(ops ExporterOptions, logger *zap.SugaredLogger) error {
 	metricsMux.Lock()
 	defer metricsMux.Unlock()
 
-	newConfig, err := createMetricsConfig(ops, logger)
-	if err != nil {
-		if curMetricsExporter == nil {
-			// Fail the process if there doesn't exist an exporter.
-			logger.Errorw("Failed to get a valid metrics config", zap.Error(err))
-		} else {
-			logger.Errorw("Failed to get a valid metrics config; Skip updating the metrics exporter", zap.Error(err))
+	if ops.StackdriverConfigMap != nil {
```

Comment:
Tests for these changes?

Reply:
This is the tests that were added to metrics/config_test.go

I placed the tests there instead of in exporter_test.go because they fit the pattern of the existing tests in config_test.go more than exporter_test.go